### PR TITLE
Chk 1904 customize v2 events

### PR DESCRIPTION
**Pull request by pietro-tota**
Depends on https://github.com/pagopa/pagopa-ecommerce-commons/pull/129

#### List of Changes
Add details for NPG payment gateway into v2 events.
Here a list of added fields compared to v1 version:

- TransactionActivatedData -> add `TransactionGatewayActivationData` interface that can be `EmptyTransactionGatewayActivationData` (PGS) or `NpgTransactionGatewayActivationData`. 
- TransactionAuthorizationCompletedData -> add `TransactionGatewayAuthorizationData`  that can be `NpgTransactionGatewayAuthorizationData` or `PgsTransactionGatewayAuthorizationData`.

Below a table with field description for each of the added data classes:

| Data class                             | Interface                           | Field                  | Gateway | Meaning                                                    |
|----------------------------------------|-------------------------------------|------------------------|---------|------------------------------------------------------------|
| EmptyTransactionGatewayActivationData  | TransactionGatewayActivationData    | none                   | PGS     | -                                                          |
| NpgTransactionGatewayActivationData    | TransactionGatewayActivationData    | orderId                | NPG     | eCommerce generated unique value for the transaction       |
| NpgTransactionGatewayActivationData    | TransactionGatewayActivationData    | correlationId          | NPG     | Correlation id unique for all NPG calls in a transaction   |
| NpgTransactionGatewayActivationData    | TransactionGatewayActivationData    | sessionId              | NPG     | Transaction sessionId (returned by NPG during build phase) |
| PgsTransactionGatewayAuthorizationData | TransactionGatewayAuthorizationData | errorCode              | PGS     | Authorization error code (if any)                          |
| PgsTransactionGatewayAuthorizationData | TransactionGatewayAuthorizationData | authorizationResultDto | PGS     | Authorization outcome                                      |
| NpgTransactionGatewayAuthorizationData | TransactionGatewayAuthorizationData | operationResult        | NPG     | Authorization outcome                                      |
| NpgTransactionGatewayAuthorizationData | TransactionGatewayAuthorizationData | operationId            | NPG     | Authorization operation id                                 |
| NpgTransactionGatewayAuthorizationData | TransactionGatewayAuthorizationData | paymentEndToEndId      | NPG     | Payment end to end id                                      |

#### Motivation and Context
Add details for NPG payment gateway into v2 events

#### How Has This Been Tested?
Junit tests

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/java/it/pagopa/ecommerce/commons/documents/v2/activation/TransactionGatewayActivationData.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package it.pagopa.ecommerce.commons.documents.v2.activation;
+
+/**
+ * Extensible interface to handle multiple payment gateway transaction activated
+ * data
+ */
+public interface TransactionGatewayActivationData {
```
Comment (giovanniberti) [inline src/main/java/it/pagopa/ecommerce/commons/documents/v2/activation/TransactionGatewayActivationData.java:7]: [design] I would make this interface `sealed`, what do you think?

File: src/main/java/it/pagopa/ecommerce/commons/documents/v2/authorization/TransactionGatewayAuthorizationData.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package it.pagopa.ecommerce.commons.documents.v2.authorization;
+
+/**
+ * Extensible interface to handle multiple payment gateway transaction outcome
+ */
+public interface TransactionGatewayAuthorizationData {
```
Comment (giovanniberti) [inline src/main/java/it/pagopa/ecommerce/commons/documents/v2/authorization/TransactionGatewayAuthorizationData.java:6]: Same here (`sealed`)

File: src/test/java/it/pagopa/ecommerce/commons/v2/TransactionTestUtils.java [hunk 3]
```diff
@@ -622,4 +628,46 @@ public static BaseTransaction reduceEvents(TransactionEvent<?>... events) {
                 );
     }
 
+    @Nonnull
+    public static TransactionGatewayAuthorizationData pgsTransactionAuthorizationCompletedData(
```
Comment (giovanniberti) [inline src/test/java/it/pagopa/ecommerce/commons/v2/TransactionTestUtils.java:632]: [naming] Why `completed`?

**Review by giovanniberti — COMMENTED** [review]
👏 Just minor nitpicks, well done!

File: src/main/java/it/pagopa/ecommerce/commons/documents/v2/activation/TransactionGatewayActivationData.java (hunk 1 above)
Comment (pietro-tota) [reply src/main/java/it/pagopa/ecommerce/commons/documents/v2/activation/TransactionGatewayActivationData.java]: Fixed with https://github.com/pagopa/pagopa-ecommerce-commons/pull/130/commits/297bb496cd320653ba116fcb1ad78fedc267f481

**Review by giovanniberti — DISMISSED** [review]
🔥

**Review by giovanniberti — APPROVED** [review]

**Pull request merged**